src/Microsoft.ML.Data/DataLoadSave/Database/DatabaseSource.cs

```diff
@@ -14,7 +14,8 @@ public sealed class DatabaseSource
         /// <param name="providerFactory">The factory used to create the <see cref="DbConnection"/>..</param>
         /// <param name="connectionString">The string used to open the connection.</param>
         /// <param name="commandText">The text command to run against the data source.</param>
-        public DatabaseSource(DbProviderFactory providerFactory, string connectionString, string commandText)
+        /// <param name="commandTimeout">The time in seconds to wait for the command to execute. Default is 30 sec.</param>
+        public DatabaseSource(DbProviderFactory providerFactory, string connectionString, string commandText, int commandTimeout = 30)
```

Comment:
This is a breaking change. You can't add a parameter (even an optional parameter) to an already shipped method.

You will need to create a new overload to add this new property.

Reply:
changing

---
In reply to: [349252694](https://github.com/dotnet/machinelearning/pull/4494#discussion_r349252694) [](ancestors = 349252694)